server/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -179,7 +179,15 @@ protected TransportReplicationAction(Settings settings, String actionName, Trans
     @Override
     protected void doExecute(Task task, Request request, ActionListener<Response> listener) {
         assert request.shardId() != null : "request shardId must be set";
-        new ReroutePhase((ReplicationTask) task, request, listener).run();
+        runReroutePhase(task, request, listener, true);
+    }
+
+    private void runReroutePhase(Task task, Request request, ActionListener<Response> listener, boolean executedByClient) {
+        try {
+            new ReroutePhase((ReplicationTask) task, request, listener, executedByClient).run();
+        } catch (Exception e) {
```

Comment:
It this necessary? `ReroutePhase` is an `AbstractRunnable` so should be fine

Reply:
Probably not? But we also do this with `new AsyncPrimaryAction(request, listener, (ReplicationTask) task).run();` in `handlePrimaryRequest`. So I did it for consistency? I think Henning was concerned that a `RuntimeException` would leave unreleased bytes.

Reply:
I did change it to just `RuntimeException`.

Reply:
ok